esp/services/ws_dfu/ws_dfuService.cpp

```diff
@@ -2083,6 +2087,13 @@ void CWsDfuEx::getFilePartsOnClusters(IEspContext &context, const char *clusterR
                 FilePart->setPartsize(partSizeStr.str());
                 if (version >= 1.38)
                     FilePart->setPartSizeInt64(size);
+                if (compressedFile && (version >= 1.58))
+                {
+                    if (compressedSize != -1)
+                        FilePart->setCompressedSize(compressedSize);
+                    else
+                        FilePart->setCompressedSize(size);
```

Comment:
It looks like this is to handle the index case (isFileKey(df)==true) ?

Is it correct though, to set both size and compressed size to same in index case? It didn't return this information previously, but Eclwatch does show both.

Reply:
Yes, this is to handle the index case (isFileKey(df)==true).
The code here is based the the existing code line 2285-2286.

Reply:
>The code here is based the the existing code line 2285-2286.

ok, so same behaviour as top-level properties, now at the part level.

It's probably ok. Indexes are always compressed, so there isn't really a separate size vs compressedSize - there could be if we wanted to calculate the uncompressed data size, but doubt we're going to do that.
So really only the compressed size is correct here for indexes, however I suspect even if it was a good idea, removing size, leaving only compressed size, would cause compatibility issues.

Leave it as it is.